### PR TITLE
Save log files for network proxy

### PR DESCRIPTION
**Pull request by Jefftree**
**What type of PR is this?**

/kind bug

**What this PR does / why we need it**:

Log artifacts for konnectivity-server and konnectivity-agent are not saved in our e2e tests (https://gcsweb.k8s.io/gcs/kubernetes-jenkins/logs/ci-kubernetes-e2e-gci-gce-network-proxy/1224163053805768711/artifacts/bootstrap-e2e-master/). We should save those files to make debugging easier

I'm not sure how we should capture konnectivity-agent since it we may not guarantee one agent per node in the future.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

/cc @caesarxuchao

### CODE REVIEW

**Comment by caesarxuchao** [comment]
/assign @dims

for approval. Thanks!

File: cluster/log-dump/log-dump.sh [hunk 1]
```diff
@@ -222,7 +222,7 @@ function export-windows-docker-event-log() {
 function save-windows-logs-via-diagnostics-tool() {
     local node="${1}"
     local dest_dir="${2}"
-    
+
```
Comment (dims) [inline cluster/log-dump/log-dump.sh]: unnecessary change?
